functions/src/merge-task-subtask.ts

```diff
@@ -0,0 +1,55 @@
+import { FirestoreCommonTask, FirestoreSubTask } from 'common/types/firestore-types';
+import { Map } from 'immutable';
+import { partition } from './util';
+import database from './db';
+
+type TaskUpdateData = { id: string; subtasks: FirestoreSubTask[] };
+
+const mergeSubtaskTask = async (): Promise<void> => {
+  const tasksSnapshot = await database.tasksCollection().get();
+  const subTasksSnapshot = await database.subTasksCollection().get();
+  // Initialize map from subtask ID to FirestoreCommonTask representing its data
+  let subTaskMap: Map<string, FirestoreSubTask> = Map();
+
+  // Populate map for constant time retrieval of subtasks by ID
+  subTasksSnapshot.forEach((document) => {
+    const data = document.data();
+    const subTask = data as FirestoreSubTask;
```

Comment:
In #635, you removed id from subtask. This cast will hide the problem that you forget to remove id from subtask.